### PR TITLE
Environment via hub 1

### PR DESCRIPTION
**Pull request by absidish**

### CODE REVIEW

File: management/server/core/hub-manager/hub-manager-impl/src/main/java/io/subutai/core/hubmanager/impl/HubEnvironmentManager.java [hunk 1]
```diff
@@ -365,9 +406,23 @@ public Integer call() throws Exception
...
+
 
     protected CompletionService<PrepareTemplatesResponseCollector> getCompletionService( Executor executor )
     {
         return new ExecutorCompletionService<>( executor );
     }
+
+
+    public RequestBuilder getSetupP2PConnectionCommand( String interfaceName, String localIp, String p2pHash,
+                                                        String secretKey, long secretKeyTtlSec )
```
Comment (Dilshat) [inline management/server/core/hub-manager/hub-manager-impl/src/main/java/io/subutai/core/hubmanager/impl/HubEnvironmentManager.java]: why do you need this command here? this is internal thing exposed by LocalPeer interface appropriate methods

File: management/server/core/hub-manager/hub-manager-impl/src/main/java/io/subutai/core/hubmanager/impl/proccessors/HubEnvironmentProccessor.java [hunk 2]
```diff
@@ -205,16 +270,184 @@ private void buildContainers( EnvironmentPeerDto peerDto )
...
+    {
+        String containerDestroyStateURL =
+                String.format( "/rest/v1/environments/%s/destroy", peerDto.getEnvironmentInfo().getId() );
+
+        LocalPeer localPeer = peerManager.getLocalPeer();
+        EnvironmentInfoDto env = peerDto.getEnvironmentInfo();
+        try
+        {
+            localPeer.cleanupEnvironment( new EnvironmentId( env.getId() ) );
+            localPeer.removeP2PConnection( env.getP2pHash() );
```
Comment (Dilshat) [inline management/server/core/hub-manager/hub-manager-impl/src/main/java/io/subutai/core/hubmanager/impl/proccessors/HubEnvironmentProccessor.java]: thi line is not needed

File: management/server/core/hub-manager/hub-manager-impl/src/main/java/io/subutai/core/hubmanager/impl/IntegrationImpl.java [hunk 3]
```diff
@@ -63,6 +70,7 @@
     private EnvironmentManager environmentManager;
     private PeerManager peerManager;
     private ConfigManager configManager;
+    private CommandExecutor commandExecutor;
```
Comment (Dilshat) [inline management/server/core/hub-manager/hub-manager-impl/src/main/java/io/subutai/core/hubmanager/impl/IntegrationImpl.java]: why do you use commandExecutor? you can call Host.execut(...)

File: management/server/core/hub-manager/hub-manager-impl/src/main/java/io/subutai/core/hubmanager/impl/HubEnvironmentManager.java [hunk 4]
```diff
@@ -176,58 +191,84 @@ public void setupVNI( EnvironmentPeerDto peerDto )
...
+
             final Future<P2PConfig> f = p2pCompletionService.take();
             P2PConfig createdConfig = f.get();
             p2pExecutor.shutdown();
-
             peerDto.setTunnelAddress( createdConfig.getAddress() );
             peerDto.setCommunityName( createdConfig.getHash() );
             peerDto.setP2pSecretKey( createdConfig.getSecretKey() );
+        }
+        catch ( ExecutionException | InterruptedException e )
```
Comment (Dilshat) [inline management/server/core/hub-manager/hub-manager-impl/src/main/java/io/subutai/core/hubmanager/impl/HubEnvironmentManager.java]: Replace 2 exception with just Exception

File: management/server/core/hub-manager/hub-manager-impl/src/main/java/io/subutai/core/hubmanager/impl/proccessors/HubEnvironmentProccessor.java [hunk 5]
```diff
@@ -162,20 +188,57 @@ private void infoExchange( final EnvironmentPeerDto peerDto )
...
+                peerDto.setSetupTunnel( true );
+            }
+            catch ( ExecutionException | InterruptedException e )
+            {
+                LOG.error( "Problems setting up tunnel", e );
+                peerDto.setSetupTunnel( false );
+            }
             updateEnvironmentPeerData( peerDto );
+        }
+        catch ( UnrecoverableKeyException | NoSuchAlgorithmException | KeyStoreException | PGPException | IOException
```
Comment (Dilshat) [inline management/server/core/hub-manager/hub-manager-impl/src/main/java/io/subutai/core/hubmanager/impl/proccessors/HubEnvironmentProccessor.java]: Just catch Exception here

**Comment by zubaidullo** [comment]
Thanks Dilshad baike. We don't know which method or which functionality can be replaced with other. B/c we're just using what we need. It would be great, to review code with SS guys.

File: management/server/core/hub-manager/hub-manager-impl/src/main/java/io/subutai/core/hubmanager/impl/HubEnvironmentManager.java (hunk 1 above)
Comment (zubaidullo) [inline management/server/core/hub-manager/hub-manager-impl/src/main/java/io/subutai/core/hubmanager/impl/HubEnvironmentManager.java]: I can't find this line in my code. It was used before

**Comment by zubaidullo** [comment]
BTW, this is not a branch, which I'm working. It's different from mine. Because, I haven't some of above lines in my branch. I have already cleaned up before.

**Pull request merged**